File: Backend/BudgetWise.Domain/Entities/Expense.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BudgetWise.Domain.Entities;

/// <summary>
/// Represents an expense.
/// </summary>
public class Expense
{
    [Key]
    public int Id { get; set; }

    [Required]
    public float Amount { get; set; }

    [Required]
    public DateTime Date { get; set; }
    
    [Required]
    public required string Name { get; set; }

    [MaxLength(255)]
    public string? Description { get; set; }

    public int UserId { get; set; }
    [ForeignKey("UserId")]
    public  User User { get; set; }
}

File: Frontend/budgetWise/src/models/expense.ts
import { User } from "./user";

export interface Expense {
    id: number;
    amount: number;
    date: string;
    name: string;
    description: string;
    userId: number;
    user: User;
}
export type ExpenseInput = {
    name: string;
    amount: number;
    date: string;
    description: string;
  };
  